amr-wind/boundary_conditions/wall_models/WallFunction.cpp

```diff
@@ -44,6 +48,17 @@ WallFunction::WallFunction(CFDSim& sim)
             (geom.ProbLo(m_direction) +
              (m_log_law.ref_index + 0.5) * geom.CellSize(m_direction));
     }
+    {
+    amrex::ParmParse pp("wave_mosd"); // "wave_mosd" is the prefix used in the input file
+    pp.query("amplitude", amplitude);
```

Comment:
instead of a new variable `amplitude` just directly do `pp.query("amplitude", m_mosd.amplitude);`. Same for the others.